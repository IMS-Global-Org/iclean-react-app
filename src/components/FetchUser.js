import React, { Component } from 'react'
import { connect } from 'react-redux'
import { validateToken } from '../actions/auth'

class FetchUser extends Component {
  defaults = { loaded: false }
  state = { ...this.defaults }

  componentDidMount = () => {
    const { isAuthenticated, dispatch } = this.props
    if( isAuthenticated ) {
      this.setAsLoaded()
    } else {
      dispatch(validateToken(this.setAsLoaded))
    }
  }

  componentWillReceiveProps = () => {
    const { loaded } = this.state
    if( !loaded ) {
      this.setAsLoaded()
    }
  }

  setAsLoaded = () => this.setState({ loaded: true })

  render = () => this.state.loaded ? this.props.children : null
}

const mapStateToProps = (state, props) => {
  return {
    isAuthenticated: state.user.id,
  }
}
export default connect(mapStateToProps)(FetchUser)
