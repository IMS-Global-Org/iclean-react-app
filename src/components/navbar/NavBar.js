import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link, withRouter } from 'react-router-dom'
import { handleLogout } from '../actions/auth'
import { Menu, Dropdown } from 'semantic-ui-react'


class NavBar extends Component {
  defaults = { activeItem: '' }
  state = { ...defaults }

  handleItemClick = ({ name: activeItem }) => this.setState({ activeItem })
}

const mapStateToProps = (state, props) => {}

export default connect(mapStateToProps)(NavBar)
