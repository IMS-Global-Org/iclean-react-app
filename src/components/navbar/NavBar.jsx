import React from 'react'
import { Menu } from 'semantic-ui-react'

import LeftNav from './LeftNav'
import RightNav from './RightNav'


const NavBar = () => (
  <Menu>
    <LeftNav />
    <RightNav />
  </Menu>
)

export default NavBar
