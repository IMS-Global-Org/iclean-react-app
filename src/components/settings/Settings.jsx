import React from 'react'
import { connect } from 'react-redux'

const Settings = () => (
  <div>
    Settings Page
  </div>
)

export default connect()(Settings)
