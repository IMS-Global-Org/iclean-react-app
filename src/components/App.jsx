import React, { Suspense, lazy } from 'react';
import { Switch, Route } from 'react-router-dom'
import ProtectedRoute from './ProtectedRoute'
import NoMatch from './NoMatch'
import FetchUser from './FetchUser'
import NavBar from './navbar/NavBar'
import Flash from './flash/Flash'

// Global Styles
import 'semantic-ui-css/semantic.min.css'

// Helper Components
import { Dimmer, Loader } from 'semantic-ui-react'

// Application HOC Components for routes
const Home = lazy(() => import ('./home/Home'))
const Settings = lazy(() => import ('./settings/Settings'))
const Login = lazy(() => import ('./Login'))
const Register = lazy(() => import ('./Register'))
const Public = lazy(() => import ('./public/Public'))


// Main Application
const App = () => (
  <React.Fragment>

    <NavBar />
    <Flash />

    <FetchUser>
      <Suspense fallback={<Dimmer active><Loader>Loading...</Loader></Dimmer>}>
        <Switch>
          <Route exact path='/' component={Home} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/register' component={Register} />
          <Route path='/public' component={Public} />
          <ProtectedRoute path='/settings/:path' component={Settings} />

          <Route component={NoMatch} />
        </Switch>
      </Suspense>
    </FetchUser>

  </React.Fragment>
)

export default App;
