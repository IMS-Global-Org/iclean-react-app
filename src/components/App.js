import React from 'react';
import { Switch, Route } from 'react-router-dom'
import ProtectedRoute from './ProtectedRoute'
import NoMatch from './NoMatch'
import FetchUser from './FetchUser'

// Application HOC Components for routes
import NavBar from './navbar/NavBar'
import Home from './home/Home'
import Settings from './settings/Settings'

// Global Styles
import 'semantic-ui-css/semantic.min.css'

// TODO create routes for main application
const App = () => (
  <React.Fragment>

    <NavBar />

    <FetchUser>
      <Switch>
        <Route exact path='/' component={Home} />
        <ProtectedRoute path='/settings' component={Settings} />

        <Route component={NoMatch} />
      </Switch>
    </FetchUser>

  </React.Fragment>
)

export default App;
