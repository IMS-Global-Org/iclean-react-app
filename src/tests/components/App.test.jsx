import React from 'react'
import { Wrapper, IndexWrapper } from '../Helpers'
import { mount, shallow } from 'enzyme'
import TestRenderer from 'react-test-renderer'
import App from '../../components/App'
import NavBar from '../../components/navbar/NavBar'
import FetchUser from '../../components/FetchUser'
import Settings from '../../components/settings/Settings'
import { Switch, Route } from 'react-router-dom'

import { BrowserRouter, MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../store'


const index = mount(
  <Wrapper>
    <App />
  </Wrapper>
)


describe('App.jsx', () => {

  describe('Creates Snapshot', () => {
    it('of the Main App', () => {
      const app = TestRenderer.create(
        <IndexWrapper>
          <App />
        </IndexWrapper>
      )

      // main application home page snapshot
      let tree = app.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })


  describe('Contains Component', () => {
    it('Navbar/Menu', () => {
      expect(index.find(NavBar)).toHaveLength(1)
    })

    it('FetchUser', () => {
      expect(index.find(FetchUser)).toHaveLength(1)
    })
  })

  describe('Has Switch Route', () => {
    xit('/ => Home', () => {
      const home = index
        .find(Route)
        .findWhere( r => r.props().path === '/')
      expect(home).toHaveLength(1)
    })
    xit('/login => Login',() => {
      const login = index
        .find(Route)
        .findWhere( r => r.props().path === '/login')
      expect(login).toBeDefined()
    })
    xit('/register => Register', () => {
      const register = index
        .find(Route)
        .findWhere( r => r.props().path === '/register')
      expect(register).toBeDefined()
    })
    xit('/logout => Logout', () => {
      const logout = index
        .find(Route)
        .findWhere( r => r.props().path === '/logout')
      expect(logout).toBeDefined()
    })
    xit('/public => Public', () => {
      const publik = index
        .find(Route)
        .findWhere( r => r.props().path === '/public')
      expect(publik).toBeDefined()
    })
    xit('/settings => Settings', () => {
      const regex = /\/settings\/:.+/
      let settings = index
        .find(Route)
        .findWhere( r => r.props().path.match(regex))
      expect(settings).toHaveLength(1)
    })
    xit(' => NoMatch', () => {
      const noMatch = index
        .find(Route)
        .findWhere( r => r.props().path === '' )
      expect(noMatch).toBeDefined()
    })
  })

})
