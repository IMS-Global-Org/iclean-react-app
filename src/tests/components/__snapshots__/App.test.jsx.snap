// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx Creates Snapshot of the Main App 1`] = `null`;
